### ISSUE TITLE
Nubeam failure with - uGNI:Device mismatch detected.  Attempting to run Gemini libraries on a(n) Aries system.

### ISSUE THREAD

**Issue by dlg0** [ORNL-Fusion/ips-atom#19]
Anybody seen this type of error before?

```
greendl1@edison04:/project/projectdirs/atom/users/greendl1/diem_tsc_error12> cat work/nb__nubeam_7/log.nubeam 
uGNI:Device mismatch detected.  Attempting to run Gemini libraries on a(n) Aries system.
uGNI:Device mismatch detected.  Attempting to run Gemini libraries on a(n) Aries system.
uGNI:Device mismatch detected.  Attempting to run Gemini libraries on a(n) Aries system.
uGNI:Device mismatch detected.  Attempting to run Gemini libraries on a(n) Aries system.
uGNI:Device mismatch detected.  Attempting to run Gemini libraries on a(n) Aries system.
uGNI:Device mismatch detected.  Attempting to run Gemini libraries on a(n) Aries system.
uGNI:Device mismatch detected.  Attempting to run Gemini libraries on a(n) Aries system.
```

I've tried two different builds of nubeam, but both give this error. Looks like a CRAY specific problem, given the Gemini and Aries (interconnect types) references. I'll probably submit a NERSC help request too.

Thanks.

**Comment by dlg0** [comment]
Looks like this may be a problem with the TRANSP build of nubeam. Using the one build from the NTCC library seems to work fine. 

@parkjm you probably want to check and switch the `mpi_nubeam_comp_exec` in ips-fastran with the one in `/project/projectdirs/atom/atom-install-edison/ntcc-gnu/bin`

**Comment by dlg0** [comment]
I think the problem is that a hopper built nubeam binary somehow ended up in the mix. Edison and hopper have different interconnects.

**Comment by parkjm** [comment]
@dlg0  I had similar error when I moved from hopper to edison with the same executable built on hopper. Nubeam worked for ncpu <= 24, so for one node, while raised similar error you had with ncpu > 24. As discussed in another post, let's build and use the common binary area.

**Comment by dlg0** [comment]
@parkjm Yep, I've but the working version in atom-install-edison/binaries/nubeam/default/

**Comment by parkjm** [comment]
@dlg0. Great!  BTW, how about using link in default/  rather than copy of binary?

**Comment by dlg0** [comment]
@parkjm I did think about that, but as we go forward we will have to maintain copies of the binaries that were available, i.e., say a new version of nubeam comes out, I'd make that the default and copy the current default to a versioned directory. Also, building the updated version likely would involve updating the source and building in the same place where the older binary was, over riding it. That would update the link, but not preserve the previous binary. Also, I think having a place where the binaries live that is separate from all the sources is less prone to the mistakes similar to what I did the other day.
